### PR TITLE
SessionService: Inject ErrorTransformer for DHE instead of Forcing Obfuscation

### PR DESCRIPTION
**Pull request by nbauernfeind**
Per request, DHE would like to decide when to obfuscate errors and have the opportunity to do more with API errors.

### CODE REVIEW

**Comment by devinrsmith** [comment]
Does this allow the ability to intercept and say, all superusers can receive unobfuscated errors, but everybody else gets obfuscated?

**Comment by nbauernfeind** [comment]
> Does this allow the ability to intercept and say, all superusers can receive unobfuscated errors, but everybody else gets obfuscated?

Yes, exactly. The AuthContext should be properly installed at each call site. I mentally audited and no calls were strict counter examples. (I make mistakes.. but I am at 95% confidence that auth context is available.)

File: server/src/main/java/io/deephaven/server/hierarchicaltable/HierarchicalTableViewSubscription.java [hunk 1]
```diff
@@ -272,14 +276,14 @@ private void process() {
                 }
             }
             if (sendError) {
-                GrpcUtil.safelyError(listener, GrpcUtil.securelyWrapError(log, upstreamFailure, Code.DATA_LOSS));
+                GrpcUtil.safelyError(listener, errorTransformer.transform(upstreamFailure));
```
Comment (devinrsmith) [inline server/src/main/java/io/deephaven/server/hierarchicaltable/HierarchicalTableViewSubscription.java:279]: I'm not sure if this context was important, but we are losing `Code.DATA_LOSS`.

Comment (nbauernfeind) [reply server/src/main/java/io/deephaven/server/hierarchicaltable/HierarchicalTableViewSubscription.java:279]: Yeah, that hurt me, too... -- But, it highlights the inability to differentiate different SRE status types when transforming non-SRE exceptions.

**Review by niloc132 — APPROVED** [review]

**Review by rcaudy — APPROVED** [review]
Code owner approval for EmbeddedServer

**Pull request merged**